### DISCUSSION TITLE
Unable to access Auth::id() from beforeCreate hook

### DISCUSSION THREAD

**Discussion by jhayiwg** [filamentphp/filament#346] [Help] [answered]
Hello,

I have a Customer Resource and I wanted to associate authenticated as a creator, so I have user_id on my customers table.

Under CreateCustomer under beforeCreate method I am trying to access authenticated user using Auth facade, auth(), Filament::auth() and request()->user() but I`m getting false or null value. Any advice.

Thank you in advance.

**Comment by ryangjchandler** [comment, marked as answer]
Use `Filament::auth()->user()` instead. Filament uses a custom guard so it handles that part for you.

**Reply by Muffinman** [reply]
I think it would be good if this was mentioned somewhere on the docs. Maybe the [users](https://filamentphp.com/docs/2.x/admin/users) page?

It seems like a pretty common requirement.

**Reply by danharrin** [reply]
This doesn't really matter anymore. When the post was created during v1, Filament had a dedicated auth guard out of the box. Now, you can just use `auth()->user()` or `Auth::user()` like a normal Laravel app.

**Reply by Muffinman** [reply]
Ah yes you're right, sorry.

I see in our case we're no longer using the default 'web' guard for the filament admin, and that's why it doesn't work OOTB for us.

**Reply by MikePonce-UwU** [reply]
Hi there, for some reason i'm not able to use this `auth()->user()` info for validating policies.
Its like, i want to know if this "product" belongs to auth user, so i check this and... i get `null`.
Maybe this is due to v3 but, i thought it doesn't have anything to do with this.

**Reply by dominikgeimer** [reply]
I have the same problem in v3

**Reply by neilangeles-philippian** [reply]
Same problem with V3

**Reply by isrardogar** [reply]
@MikePonce-UwU have you found solution ?

**Reply by danharrin** [reply]
Open an issue with a reproduction repository if you think it's Filament. But I can almost guarantee it's going to be something on your end. So please create a *new* repository and replicate the issue instead of using your existing app.

**Reply by elchinabilov** [reply]
Any way to fix the problem?

**Reply by danharrin** [reply]
Locked as someone just needs to create a reproduction repository if it's a bug.
